xarray/backends/netCDF4_.py

```diff
@@ -106,6 +106,7 @@ def _getitem(self, key):
             with self.datastore.lock:
                 original_array = self.get_array(needs_lock=False)
                 array = getitem(original_array, key)
+                array = np.asarray(array)
```

Comment:
@shoyer does this look right? (also see above for h5netcdf)

Reply:
Yes, I would definitely put the comment here, too, though!

Reply:
Hmmm.. this is also breaking the backend contract. Previously it was OK to return scalars from `__getitem__`